src/tests/rack-resize.test.ts

```
@@ -0,0 +1,118 @@
+  describe("canResizeRackTo", () => {
+    it("allows growing the rack regardless of contents", () => {
+      const device = createTestDevice({ position: 40 });
+      const rack = createTestRack({ height: 24, devices: [device] });
+      const dt = createTestDeviceType({
+        slug: device.device_type,
+        u_height: 1,
+      });
+
+      const result = canResizeRackTo(rack, 42, [dt]);
+      expect(result.allowed).toBe(true);
+      expect(result.conflicts).toHaveLength(0);
```

ESLint

[error] Avoid exact length assertions on data arrays (breaks on additions). Use .length > 0 for existence checks. For behavioral invariants (deduplication, pagination), use eslint-disable-next-line with justification.

(no-restricted-syntax)